### PR TITLE
ENH: Show column name in assert_frame_equal

### PR DESCRIPTION
**Pull request by moi90**
The output of the AssertionError raised by assert_frame_equal is changed:

- old: `AssertionError: Attributes of DataFrame.iloc[:, 1] are different`
- new: `AssertionError: Attributes of DataFrame.loc[:, 'b'] are different`

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
hmm, the only issue is this doesn't quite work for duplicated names, but i guess this is friendlier for readers of the error message.

**Comment by jreback** [comment]
cc @TomAugspurger @jorisvandenbossche

**Comment by TomAugspurger** [comment]
Yeah duplicates labels was my initial concern.

Can we update the error message to keep using `.iloc[:, i]`, but also include `column name=" "` somewhere?

**Comment by moi90** [comment]
Thanks for the feedback!
I see, I didn't think about duplicate names (and didn't know that they are allowed at all).

**Comment by moi90** [comment]
Sorry for the multiple commits. Feel free to squash.

**Comment by jbrockmendel** [comment]
@moi90 no need to worry about the squashing, but we do need to get the tests passing.  looks like there are a handful of places where you need to update the expected error message

**Comment by moi90** [comment]
I will look into this.

**Comment by alimcmaster1** [comment]
@moi90 any luck? If you have questions feel free to post them on there :)

File: pandas/util/testing.py [hunk 1]
```diff
@@ -1400,7 +1400,9 @@ def assert_frame_equal(
                 check_names=check_names,
                 check_datetimelike_compat=check_datetimelike_compat,
                 check_categorical=check_categorical,
-                obj="{obj}.iloc[:, {idx}]".format(obj=obj, idx=i),
+                obj='{obj}.iloc[:, {idx}] (column name="{col}")'.format(
```
Comment (WillAyd) [inline pandas/util/testing.py]: If you wanted to could just use f-strings to make this even more compact

**Review by WillAyd — COMMENTED** [review]
Looks good to me. Optional comment from my perspective

File: pandas/util/testing.py (hunk 1 above)
Comment (moi90) [reply pandas/util/testing.py]: They are new in 3.6. Is anything older not supported anymore by pandas?

Comment (jreback) [reply pandas/util/testing.py]: no we just dropped 3.5

**Comment by jreback** [comment]
@moi90 if you can use f-strings and ping on green.

**Comment by jbrockmendel** [comment]
@moi90 can you rebase

File: pandas/util/testing.py [hunk 2]
```diff
@@ -2389,17 +2356,149 @@ def wrapper(*args, **kwargs):
...
-                )
+                skip("Skipping test due to lack of connectivity" f" and error {err}")
 
     return wrapper
 
 
 with_connectivity_check = network
 
 
+def assert_raises_regex(_exception, _regexp, _callable=None, *args, **kwargs):
```
Comment (WillAyd) [inline pandas/util/testing.py]: This was removed in #29174 so want to be careful not to add back in here

**Review by WillAyd — CHANGES_REQUESTED** [review]
This looks close. Looks like maybe an issue on previous conflict fix ups; if you can resolve and repush would be great!

**Comment by jbrockmendel** [comment]
the CI error in the linter usually means there is an unused import somehwere

**Comment by moi90** [comment]
No, it's an error in flake8:

```
   File "/home/runner/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/flake8/formatting/default.py", line 39, in format
    "col": error.column_number,
ValueError: unsupported format character ':' (0x3a) at index 41
```

I'm beginning to give up... Can't someone of the core team bring it to a successful conclusion?

**Comment by jbrockmendel** [comment]
```
ValueError: unsupported format character ':' (0x3a) at index 41
```

The last few times I've seen this, it turned out there was an unused import somewhere.

**Comment by jreback** [comment]
can you merge master

**Comment by jorisvandenbossche** [comment]
@moi90 I merged master and fixed some flake8 issues. Not sure why it failed before, but know that you can always run the linting checks locally (I recommend installing git commit hooks for this: https://dev.pandas.io/docs/development/contributing.html#python-pep8-black).

All green now, so will merge. Thanks for the PR!

**Comment by moi90** [comment]
Wonderful, thanks! And you're welcome :)

**Pull request merged**